homeassistant/components/tcp.py

```diff
@@ -0,0 +1,125 @@
+"""
+homeassistant.components.tcp
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+A generic TCP socket component.
+"""
+import logging
+import socket
+import re
+from select import select
+
+from homeassistant.const import CONF_NAME, CONF_HOST
+from homeassistant.helpers.entity import Entity
+
+
+DOMAIN = "tcp"
+
+CONF_PORT = "port"
+CONF_TIMEOUT = "timeout"
+CONF_PAYLOAD = "payload"
+CONF_UNIT = "unit"
+CONF_VALUE_REGEX = "value_regex"
```

Comment:
I don't know how I'd use a template in this instance. For example, if you want say, the boiler status from my boiler you send `payload` of:
`r Status01\n`
from which you receive a response of:
`54.0;44.5;-;-;-;on`
I'm only interested in the 'on' part, so I use the following regex to extract it:
`.+;(\w+)`
How would I do that with a template?